### PR TITLE
Version 2.2.6 - Fixing showbox once again

### PR DESCRIPTION
**Pull request by JorrinKievit**
- #126 
- #127 
- #131

### CODE REVIEW

File: .eslintrc
Comment (binaryoverload) [inline .eslintrc:1]: How come we've moved from a .js file?

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -11,28 +11,33 @@ jobs:
...
-      run: npm install
-    
-    - name: Build project
-      working-directory: ./.docs
-      run: npm run generate
-      env:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - uses: pnpm/action-setup@v2
```
Comment (binaryoverload) [inline .github/workflows/docs.yml:17]: action-setup is now on v3

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -13,17 +13,22 @@ jobs:
     steps:
       - name: Checkout code
         uses: actions/checkout@v3
-      
+
+      - uses: pnpm/action-setup@v2
```
Comment (binaryoverload) [inline .github/workflows/publish.yml:17]: Same as above

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -13,22 +13,30 @@ jobs:
...
-    - name: Checkout code
-      uses: actions/checkout@v3
+      - name: Checkout code
+        uses: actions/checkout@v3
 
-    - name: Install Node.js
-      uses: actions/setup-node@v3
-      with:
-        node-version: 18
+      - uses: pnpm/action-setup@v2
```
Comment (binaryoverload) [inline .github/workflows/tests.yml:19]: Same as above

File: src/__test__/providers/embedUtils.ts [hunk 4]
```diff
@@ -1,12 +1,12 @@
...
-import { makeSimpleProxyFetcher } from "@/fetchers/simpleProxy";
+import { buildProviders } from '@/entrypoint/builder';
+import { ScrapeMedia } from '@/entrypoint/utils/media';
+import { targets } from '@/entrypoint/utils/targets';
+import { makeStandardFetcher } from '@/fetchers/standardFetch';
+import { Embed, Sourcerer, SourcererEmbed } from '@/providers/base';
+import { TestTypes } from './providerUtils';
+import { describe, expect, it } from 'vitest';
+import { ProviderControls } from '@/entrypoint/controls';
+import { makeSimpleProxyFetcher } from '@/fetchers/simpleProxy';
```
Comment (binaryoverload) [inline src/__test__/providers/embedUtils.ts:9]: Not sure I like the change from double to single quotes but I'll allow it

File: src/utils/valid.ts [hunk 5]
```diff
@@ -15,3 +17,54 @@ export function isValidStream(stream: Stream | undefined): boolean {
   // unknown file type
   return false;
 }
+
+export async function validatePlayableStream(
+  stream: Stream,
+  ops: ProviderRunnerOptions | IndividualEmbedRunnerOptions,
+): Promise<Stream | null> {
+  const fetcher = stream.flags.length === 1 && stream.flags.includes('cors-allowed') ? ops.fetcher : ops.proxiedFetcher;
```
Comment (binaryoverload) [inline src/utils/valid.ts:25]: How come we're checking the length as well as includes? Just an optimisation?

File: src/utils/valid.ts [hunk 6]
```diff
@@ -15,3 +17,54 @@ export function isValidStream(stream: Stream | undefined): boolean {
...
+            ...stream.preferredHeaders,
+            ...stream.headers,
+          },
+        }),
+      ),
+    );
+    // remove invalid qualities from the stream
+    const validQualities = stream.qualities;
+    Object.keys(stream.qualities).forEach((quality, index) => {
+      if (validQualitiesResults[index].statusCode !== 200) {
```
Comment (binaryoverload) [inline src/utils/valid.ts:52]: Do the fetchers not return exceptions in the case of a non-success code?

If they do, we should use `Promise.allSettled`

**Review by binaryoverload — CHANGES_REQUESTED** [review]

File: src/utils/valid.ts (hunk 5 above)
Comment (JorrinKievit) [reply src/utils/valid.ts:25]: In case of [CORS_ALLOWED, IP_LOCKED] 😬

File: .eslintrc
Comment (JorrinKievit) [reply .eslintrc:1]: The vite eslint plugin was complaining about it being commonjs so I figured this was the easiest solution 😂

File: src/__test__/providers/embedUtils.ts (hunk 4 above)
Comment (JorrinKievit) [reply src/__test__/providers/embedUtils.ts:9]: Has been the config since the prettier config was committed 8 months ago

Comment (binaryoverload) [reply src/__test__/providers/embedUtils.ts:9]: Sus, I think this is a quota saving message by big quote

**Review by binaryoverload — APPROVED** [review]

**Pull request merged**